guides/v2.2/ui_comp_guide/concepts/ui_comp_config_flow_concept.md

```diff
@@ -39,8 +39,10 @@ When the request for my_page comes, the server does the following:
 
 1. Determines which UI components are used in this particular layout. In the example, the UI components that are used are defined in the `my_form` component’s `.xml` declaration file.
 2. Searches the `.xml` files with name `my_form` among all modules. The server then merges all the `my_form .xml` file(s) into a single configuration object, thus overriding the common properties, so that the latest `my_form .xml` file always has the highest priority.
-1. Merges the resulting configuration (from Step 2 above) with the configuration from the UI module `definition.xml`. The UI module `definition.xml` configuration file has the lowest priority, and is overwritten by the merged configuration of all `my_form.xml` files.
-2. Translates the resulting configuration into JSON format and adds it to response body the following way:
+3. Merges the resulting configuration (from Step 2 above) with the configuration from the UI module `definition.xml`. The UI module `definition.xml` configuration file has the lowest priority, and is overwritten by the merged configuration of all `my_form.xml` files.
+4. Translates the resulting configuration into JSON format.
```

Comment:
Actually, one of the last things Magento does is convert the configuration to JSON. The `modifyMeta()` and `modifyData()` methods of a DataModifier are given complex PHP multidimensional arrays to work with. After all of the modifications have been done, the component data is convert to JSON and injected to the page via a `__toString()` method. This is where it all goes down: `\Magento\Ui\TemplateEngine\Xhtml\Result::appendLayoutConfiguration()`. Perhaps this point, `4.`, should be reversed with `5.`? Perhaps to help others debug, it might be a good idea to even reference the class and function?

Reply:
Hey @siliconrockstar, just wanted to follow up with you to see if you can revise this per @bassplayer7 suggestion so we can get these changes integrated to production. thanks.